[Chua Gene] iP

iamgenechua opened the pull request.

Commit:
Add Gradle support

iamgenechua committed:
Implement Kai's ability to greet, echo and say bye to user

Program showcases Kaizen logo upon startup.
Kai will greet user automatically and echo what the user types.
Kai will only exit when user types in 'Bye'.
Interface includes different colors and speech prefaces to allow for
easier identification of who the messages belong to.

iamgenechua committed:
Add a todoList for user to key data into

An arraylist is used to store the toDoList so users can store as much
data as they like

iamgenechua committed:
Change Interactor to a non-static class

To avoid the problem of static method having to access
non-static variables.

iamgenechua committed:
Create add method and abstract out bye message

iamgenechua committed:
Add showList method to show user his To-do list

List is shown by iterating through every item in the arraylist

iamgenechua committed:
Change angular brackets to square brackets in add methodd

For cosmetic purposes

iamgenechua committed:
Generate JavaDocs in the docs folder

iamgenechua committed:
Change access modifiers of ChatBot fields to private.

iamgenechua committed:
Improve documentation for showList() and add() method

iamgenechua committed:
Create a new Task class

iamgenechua committed:
Implement status check of tasks upon checking of list

iamgenechua committed:
Implement marking of task as done

iamgenechua committed:
Create ToDo: child class of Task class

iamgenechua committed:
Add tolowercase() to reduce input bugs

To reduce bugs arising from case-sensitive issues of input.

iamgenechua committed:
Override ToDo methods to return ToDo instead of Task

Implemented checking to check if an item is a ToDo and to return
the same class after checking.

iamgenechua committed:
Add Deadline class as a child class of Task

iamgenechua committed:
Implement and furnish Event class

iamgenechua committed:
Remove colors to avoid unnecessary complications

iamgenechua committed:
Semi-automate Text UI Testing

iamgenechua committed:
Handle errors related to makeTaskDone

Catch errors arising when:
- user types in nothing after 'done'
- user types in a non-integer
- user types in a taskNumber that's not within the list range

iamgenechua committed:
Refactor the getInput() method to easily process command from user

iamgenechua committed:
Separate todo, deadline and event into separate methods

iamgenechua committed:
Update Text-UI-Test to handle more exceptions

iamgenechua committed:
Add delete functionality

iamgenechua committed:
Add Enums as commands

Changed switch case to allow for Enums to be used

iamgenechua committed:
Change Java JDK to 11

iamgenechua committed:
Edit testing input and expected output

iamgenechua committed:
Create Storage class to store task list

iamgenechua committed:
Edit modifier access level of chatBot from private to default

To allow for easier writing and reading of harddisk files

iamgenechua committed:
Abstract out TaskList class

To allow for easier loading and writing of data between Storage and TaskList

iamgenechua committed:
Implement writeToFile method in Storage class

iamgenechua committed:
Rename chatBot to Ui inline with iP task

iamgenechua committed:
Implement WriteToFile() for all user actions

iamgenechua committed:
Implement Read and Write to file successfully

Also, refactored out the code loop to Duke

iamgenechua committed:
Implement loading and writing files successfully

Also, refactored looping code onto Duke.java

iamgenechua committed:
Commit to save

iamgenechua committed:
Implement deadline date and time parsing and formatting

iamgenechua committed:
Commit data/duke.txt files before checkout

iamgenechua committed:
Merge branch 'branch-Level-7' to include save feature

iamgenechua committed:
Merge branch 'branch-Level-8'. Resolve conflicts.

iamgenechua committed:
Add some Javadocs

iamgenechua committed:
Add Ui classes and functionality

iamgenechua committed:
Extract Storage class

iamgenechua committed:
Extract Parser class

iamgenechua committed:
Extract TaskList

iamgenechua committed:
Put classes into one duke package

iamgenechua committed:
Add JUnit tests

More than two test methods added, targeting 3 different classes

iamgenechua committed:
Add Jar File

iamgenechua committed:
Add Clear Command

iamgenechua committed:
Add Clear Command to clear all tasks

iamgenechua committed:
Add all changes in text-ui-test

iamgenechua committed:
Update runtest.sh with clear command

iamgenechua committed:
Add JavaDoc comments to all non-trivial methods

iamgenechua committed:
Follow good coding style

Deleted some unnecessary white spaces

iamgenechua committed:
Implement Find function

iamgenechua committed:
Merge branch 'branch-A-JavaDoc'

iamgenechua committed:
Merge branch 'branch-A-CodingStandard'

iamgenechua committed:
Merge branch 'branch-Level-9'

iamgenechua committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

iamgenechua committed:
Add Gradle and Checkstyle plugin successfully

iamgenechua committed:
Create Jar file with Gradle successfully

iamgenechua committed:
Merge branch 'branch-A-Gradle'

iamgenechua committed:
Detect and correct coding style violations

iamgenechua committed:
Clear all coding violations so gradle can build succesfully

iamgenechua committed:
Merge branch 'branch-A-Checkstyle'

iamgenechua committed:
Add Launcher class and Duke extends Application

iamgenechua committed:
Refactor code to make GUI

iamgenechua committed:
Fix text height in GUI

iamgenechua committed:
Fix CheckStyle

iamgenechua committed:
Merge branch 'branch-Level-10'

iamgenechua committed:
Merge to master branch

Change mainClassName in gradle to duke.Launcher

munharsha commented on src/main/java/duke/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,92 @@
+
+    /**
+     * Adds a task to the tasklist.
+     * @param tasks the tasklist containing tasks so far
```
For the javadocs, there should be a spacing between description and @param

iamgenechua replied:
Wow great catch! You are most definitely right.

munharsha commented on src/main/java/duke/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,92 @@
+        if (descriptionArray.length != 2) {
+            throw new DukeException("NANI??! Enter your deadline name, date & time(if any) (e.g. 2020-01-30 18:30)!\n");
+        }
+
```
This execute function is quite long. Maybe can break up further into parts for better readability?

iamgenechua replied:
Thank you for your constructive comment! I will get to it.

munharsha commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,46 @@
+ * Encapsulates an Abstract Command class
+ */
+public abstract class Command {
+    private String[] parsedCommand;
```
Maybe this can be made protected as there are classes that extend from it so they can directly access this variable instead of using a getter.

iamgenechua replied:
Yes very good idea thank you.

munharsha commented on src/main/java/duke/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+        ui.appendMessage("Here are the matching tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i)
+                    .getDescription()
```
Maybe can break this block up as there is a lot going on this one block. It was quite difficult to trace through.

iamgenechua replied:
Yes good suggestion.

munharsha commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+     * Check if duke.TaskList is empty.
+     * @return if taskList is empty or not
+     */
+    boolean isEmpty() {
```
Should be consistent with naming functions for example adding the public

munharsha reviewed (commented):
Overall your code looks really clean and you have mostly followed the java conding standards. I liked how you named your variables and functions.

iamgenechua reviewed (commented):
Good constructive comments that I am going to work on.

iamgenechua committed:
Assert that .clearMessage() works as intended

iamgenechua committed:
Remove assert

Need to checkout to a new branch first

iamgenechua committed:
Assert that .clearMessage() works as intended

Assert that ui message is empty at the start of each getResponse method call

iamgenechua committed:
Assert that the todo description is not empty

iamgenechua committed:
Assert that todo descriptions are not empty or blank

Also now the bot throws exception error when user keys in decription filled with spaces

iamgenechua committed:
Assert and handle exceptions for deadline command

Throw error if deadline name is empty
Assert that the deadline is not blank

iamgenechua committed:
Refactor Find Command for easier readability

Break up chained methods into multiple shorter lines of code

iamgenechua committed:
Change access modifier of parsedCommand variable in Command Class

Changed from private to protected since child classes can access it without getter methods

iamgenechua committed:
Merge pull request #3 from iamgenechua/branch-A-Assertions

Branch a-assertions

iamgenechua committed:
Remove additional semi-colon in AddToDoCommand Class

iamgenechua committed:
Merge branch 'master' into branch-A-CodeQuality

iamgenechua committed:
Merge pull request #4 from iamgenechua/branch-A-CodeQuality

Branch a-code quality

iamgenechua committed:
Implement flexible search algorithm for tasks

iamgenechua committed:
Improve some code quality

iamgenechua committed:
Merge branch 'branch-C-BetterSearch'

iamgenechua committed:
Add friendlier syntax for TODO

Users can just type in 't' to make Todos

iamgenechua committed:
Add friendlier syntax for deadline and event

Users can now use 'd' for deadline
Users can now use 'e' for event

iamgenechua committed:
Add friendlier syntax for list

Users can now type in 'ls' to show the list

iamgenechua committed:
Merge branch 'branch-C-FriendlierSyntax'

iamgenechua committed:
Fix bug pertaining to retrieval of task from storage

iamgenechua committed:
Set up CI

iamgenechua committed:
Clean up JavaDocs Comments

iamgenechua committed:
Add UI.png

iamgenechua committed:
Write User Guide

iamgenechua committed:
Correctly named Ui.png file

iamgenechua committed:
Set theme jekyll-theme-modernist

iamgenechua committed:
Enable welcome message

iamgenechua committed:
Merge branch 'master' of https://github.com/iamgenechua/ip

iamgenechua committed:
Fix JUnit Test for windows Computer

iamgenechua committed:
Edit README

iamgenechua committed:
Update docs User Guide

iamgenechua committed:
Update README.md
